tedge_config/src/tedge_config.rs

```diff
@@ -42,29 +43,19 @@ impl ConfigSettingAccessor<DeviceIdSetting> for TEdgeConfig {
     }
 }
 
-impl ConfigSettingAccessor<WritableDeviceIdSetting> for TEdgeConfig {
-    fn query(&self, _setting: WritableDeviceIdSetting) -> ConfigSettingResult<String> {
-        self.data
-            .device
-            .id
-            .clone()
-            .ok_or(ConfigSettingError::ConfigNotSet {
-                key: WritableDeviceIdSetting::KEY,
-            })
-    }
-
-    fn update(
-        &mut self,
-        _setting: WritableDeviceIdSetting,
-        value: String,
-    ) -> ConfigSettingResult<()> {
-        self.data.device.id = Some(value);
-        Ok(())
-    }
-
-    fn unset(&mut self, _setting: WritableDeviceIdSetting) -> ConfigSettingResult<()> {
-        self.data.device.id = None;
-        Ok(())
+fn cert_error_into_config_error(key: &'static str, err: CertificateError) -> ConfigSettingError {
+    match &err {
+        CertificateError::IoError(io_err) => match io_err.kind() {
+            std::io::ErrorKind::NotFound => ConfigSettingError::ConfigNotSet { key },
+            _ => ConfigSettingError::DerivationFailed {
+                key,
+                cause: format!("{}", err),
```

Comment:
There is no error or waning in real. However, IDE complains  ```CertificateError` doesn't implement `Display` (required by {})```. I don't understand why it works.

Reply:
Indeed, Intelij Idea erroneously reports an error here, while there is none.

* In that case, it seems because the `Display` trait is implicitly derived by `#[derive(thiserror::Error)]`.
* There are also false positives, when Idea is happy but Rustc is not.

Reply:
I didn't know that the `Display` trait is derived by `#[derive(thiserror::Error)]`. Good to know. Thanks.